app/assets/javascripts/pub_sub.js

```diff
@@ -0,0 +1,16 @@
+(function($) {
+  var o = $({});
+
+  $.subscribe = function() {
+    o.on.apply(o, arguments);
+  };
+
+  $.unsubscribe = function() {
+    o.off.apply(o, arguments);
```

Comment:
Exactly right. To someone reading the code, `subscribe` and `publish` immediately invoke the pub sub pattern. `on` and `trigger` are quite jquery specific.